roles/jd-server/src/lib/job_declarator/mod.rs

```diff
@@ -231,9 +222,54 @@ impl JobDeclaratorDownstream {
+                        // 2. there are some unknown txids. Just before sending PMT, the JDS
+                        //    mempool is triggered to fill the known txids with the full
+                        //    transactions. When a PMTS arrives, just before sending a DMJS, the
+                        //    unknown full transactions provided by the downstream are added to the
+                        //    JDS mempool
                         match next_message_to_send {
-                            Ok(SendTo::Respond(message)) => {
-                                Self::send(self_mutex.clone(), message).await.unwrap();
+                            Ok(SendTo::Respond(m)) => {
+                                match m {
+                                    JobDeclaration::AllocateMiningJobToken(_) => {
+                                        error!("Send unexpected message: AMJT")
+                                    }
+                                    JobDeclaration::AllocateMiningJobTokenSuccess(_) => {
+                                        info!("Send message: AMJTS")
+                                    }
+                                    JobDeclaration::DeclareMiningJob(_) => {
+                                        error!("Send unexpected message: DMJ");
+                                    }
+                                    JobDeclaration::DeclareMiningJobError(_) => {
+                                        info!("Send nmessage: DMJE")
+                                    }
+                                    JobDeclaration::DeclareMiningJobSuccess(_) => {
+                                        info!("Send message: DMJS. Updating the JDS mempool.");
+                                        Self::send_txs_to_mempool(self_mutex.clone()).await;
+                                    }
+                                    JobDeclaration::IdentifyTransactions(_) => {
+                                        info!("Send  message: IT")
+                                    }
+                                    JobDeclaration::IdentifyTransactionsSuccess(_) => {
+                                        error!("Send unexpected message: ITS")
+                                    }
+                                    JobDeclaration::ProvideMissingTransactions(_) => {
+                                        info!("Send message: PMT. Updating the JDS mempool.");
+                                        Self::send_txs_to_mempool(self_mutex.clone()).await;
+                                    }
+                                    JobDeclaration::ProvideMissingTransactionsSuccess(_) => {
+                                        error!("Send unexpected PMTS");
+                                    }
+                                    JobDeclaration::SubmitSolution(_) => todo!(),
+                                }
+                                Self::send(self_mutex.clone(), m).await.unwrap();
```

Comment:
this always send the message also for the error cases

Reply:
if we get one of the error cases above something really strange is happening better to close the connection. Downstream will decide what to do